cpp/assetmanager.cpp

```diff
@@ -43,7 +33,7 @@ bool AssetManager::can_load(const std::string &name) const {
 Texture *AssetManager::load_texture(const std::string &name) {
 	std::string filename = this->root->join(name);
 
-	Texture *ret;
+	tex_ptr ret(nullptr, [](Texture *t) { delete t; });
```

Comment:
It can but here is the constructor call explicitly noted what is better style imho.